### PR TITLE
Fix/webdriver element and wdio-ui5-key

### PR DESCRIPTION
**Pull request by dominikfeininger**

### CODE REVIEW

File: lib/wdioUi5-index.js [hunk 1]
```diff
@@ -244,29 +252,75 @@ function setup(context) {
...
 
     /**
-     * this function is the main method to interact with the UI5 application
+     * creates a string valid as object key from a selector
+     * @param {sap.ui.test.RecordReplay.ControlSelector} selector
+     * @returns {String} wdio_ui5_key
+     */
+    function createWdioUI5KeyFromSelector(selector) {
+
+        var orEmpty = function (string) {
```
Comment (vobu) [inline lib/wdioUi5-index.js:261]: `const orEmpty = (string) => { ... }`

File: lib/wdioUi5-index.js [hunk 2]
```diff
@@ -244,29 +252,75 @@ function setup(context) {
...
+        return wdi5_ui5_key;
+    };
+
+    /**
+     * to generate an object key from any string
+     * @param {String} key
+     * @returns {String}
+     */
+    function stripNonValidCharactersForKey(key) {
+        key = key.split('.').join('');
```
Comment (vobu) [inline lib/wdioUi5-index.js:274]: concatenate the chain à la 
```javascript
key.split().join()
.split().join()
.split().join()
//...
.toLowerCase()
```
for better readability

File: .vscode/launch.json [hunk 3]
```diff
@@ -24,6 +24,23 @@
             "type": "node",
             "program": "${workspaceFolder}/node_modules/.bin/wdio",
             "request": "launch",
+            "args": [
+                "${workspaceFolder}/test/wdio-browser.conf.js",
+                "--spec",
+                "test-ui5-ad.js"
```
[the remark this answers is not in the document]
Comment (dominikfeininger) [reply .vscode/launch.json:30]: have two task one with static, one with ${file}. Renamed the tasks for better differentiation

**Comment by dominikfeininger** [comment]
marked wdio-ui5-key as optional in Readme.md

File: lib/wdioUi5-index.js [hunk 4]
```diff
@@ -244,29 +252,75 @@ function setup(context) {
...
 
     /**
-     * this function is the main method to interact with the UI5 application
+     * creates a string valid as object key from a selector
+     * @param {sap.ui.test.RecordReplay.ControlSelector} selector
+     * @returns {String} wdio_ui5_key
+     */
+    function createWdioUI5KeyFromSelector(selector) {
+
+        var orEmpty = (string) => {
```
Comment (vobu) [inline lib/wdioUi5-index.js:261]: `var` -> `const`

File: lib/wdioUi5-index.js [hunk 5]
```diff
@@ -244,29 +252,75 @@ function setup(context) {
...
+        return wdi5_ui5_key;
+    };
+
+    /**
+     * to generate an object key from any string
+     * @param {String} key
+     * @returns {String}
+     */
+    function stripNonValidCharactersForKey(key) {
+        key = key.split('.').join('')
```
Comment (vobu) [inline lib/wdioUi5-index.js:274]: `return key.split()....`

File: lib/wdioUi5-index.js [hunk 6]
```diff
@@ -244,29 +252,75 @@ function setup(context) {
...
+     * wdio_ui5_key is generated based on the givven selector.
+     * If wdio_ui5_key is provided with the selector the provided wdio_ui5_key is used.
+     * You can force to load the control freshly from browser context by setting the 'forceSelect' parameter to true
+     *
      * @param {WDI5Selector} wdi5Selector custom selector object with property wdio_ui5_key and sap.ui.test.RecordReplay.ControlSelector
      */
-    context.addCommand('asControl', (wdi5Selector) => {
-        // facade on a UI5 control
-        // to expose certain methods of the UI5 control via wdio
+    context.addCommand('asControl', async (wdi5Selector) => {
```
Comment (vobu) [inline lib/wdioUi5-index.js:314]: why flagged as `async` here? there doesn't seem to be actual async (`await` or such) code in here

File: test/ui5-app/webapp/test/e2e/test-ui5-ad.js [hunk 7]
```diff
@@ -2,24 +2,44 @@ const assert = require('assert');
...
+    const viewName = 'test.Sample.view.Main';
 
-    it('check the binding of the username input', () => {
+    it('check the binding of the username input with custom wdio_ui5_key', () => {
         // set new Username
         const newUsername = 'my New Username';
 
         // create selector
         const inputSelector = {
+            force: true,
```
Comment (vobu) [inline test/ui5-app/webapp/test/e2e/test-ui5-ad.js:13]: `forceSelect`?

**Pull request merged**